git-sync/ssh.md

```diff
@@ -0,0 +1,64 @@
+# Using SSH with git-sync
+
+Git-sync supports using the SSH protocol for pulling Git content.
+
+For now, only keys without password protection are supported.
+
+## Setup
+
+Write a config file for a Secret with name "git-creds" that holds your SSH private key, with the key mapped to "id-rsa".
+```
+{
+  "kind": "Secret",
+  "apiVersion": "v1",
+  "metadata": {
+    "name": "git-creds"
+  },
+  "data": {
+    "id-rsa": <private-key>
+}
+```
+
+Create the Secret.
+```
+kubectl create -f /path/to/secret-config.json
+```
+
+In your Pod or Deployment configuration, specify a Secret volume to contain the "git-creds" secret.
+```
+volumes: [
+    {
+        "name": "git-secret",
+        "secret": {
+          "secretName": "git-creds"
+        }
+    },
+    ...
+],
+```
+
+In your git-sync container configuration, mount the Secret volume at "/etc/git-secret". Ensure that the environment variable GIT_SYNC_REPO is set to use a URL with the SSH protocol, and set GIT_SYNC_SSH to true.
+```
+{
+    name: "git-sync",
+    ...
+    env: [
+        {
+            name: "GIT_SYNC_REPO",
+            value:  "git@github.com:kubernetes/kubernetes.git",
+        }, {
+            name: "GIT_SYNC_SSH",
+            value: "true",
+        },
+    ...
+    ]
+    volumeMounts: [
+        {
+            "name": "git-secret",
+            "mountPath": "/etc/git-secret"
+        },
+        ...
+    ],
+}
+```
+**Note: Do not mount the secret volume with "readOnly: true".** Due to a peculiarity with permissions, "readOnly: true" makes the Secret readable for user, group, and others (0444), which is not restrictive enough to be used as an SSH key, yet the read-only nature prevents changing the permissions. Omitting readOnly allows for modifying the permissions from within the container.
```

Comment:
I wasn't sure how to catch the readOnly attribute itself, but if it is set, then os.Chmod will return an error with a warning about this.

```
// Secret permissions must be restricted to use it as SSH key
	if err := os.Chmod("/etc/git-secret/id-rsa", 0400); err != nil {

		// Need to make sure secret volume isn't mounted readOnly, which gives default permission 0444
		// 0444 is not restrictive enough to use as SSH key, but also prevents chmod
		return fmt.Errorf("error running chmod on secret (make sure secret volume is NOT mounted with readOnly=true): %v", err)
	}
```